src/main/java/nl/dtls/fairdatapoint/api/controller/metadata/GenericMetaController.java

```diff
@@ -87,17 +92,35 @@ public MetaDTO getMeta(HttpServletRequest request) throws MetadataServiceExcepti
         // 3. Get and check existence entity
         IRI uri = i(getRequestURL(request, persistentUrl));
         IRI entityUri = removeLastPartOfIRI(uri);
-        Model model = metadataService.retrieve(entityUri);
+        Model entity = metadataService.retrieve(entityUri);
 
         // 4. Get member
-        String entityId = getMetadataIdentifier(model).getIdentifier().getLabel();
+        String entityId = getMetadataIdentifier(entity).getIdentifier().getLabel();
         Optional<MemberDTO> oMember = memberService.getMemberForCurrentUser(entityId, Metadata.class);
         MemberDTO member = oMember.orElse(new MemberDTO(null, null));
 
         // 5. Get state
-        MetaStateDTO state = metadataStateService.getState(entityUri, model, rd);
+        MetaStateDTO state = metadataStateService.getState(entityUri, entity, rd);
+
+        // 6. Make path map
+        HashMap<String, MetaPathDTO> pathMap = new HashMap<>();
+        while (true) {
+            MetaPathDTO entry = new MetaPathDTO();
+            entry.setResourceDefinitionUuid(rd.getUuid());
+            entry.setTitle(getTitle(entity).stringValue());
+            IRI parentUri = i(getStringObjectBy(entity, entityUri, DCTERMS.IS_PART_OF));
+            entry.setParent(Optional.ofNullable(parentUri).map(IRI::toString).orElse(null));
```

Comment:
Any options for a non-explicit `null`? Not sure what the best approach would be, possibly something like
```java
Optional.ofNullable(parentUri)
    .map(IRI::toString)
    .ifPresent(entry::setParent);
```

Depending on how MetaPathDto handles absent values.

Reply:
True... thanks, will do.